### DISCUSSION TITLE
Reduce image before uploading.

### DISCUSSION THREAD

**Discussion by ghost** [ryelogheat/xpbot#40] [Ideas]
Hi. Is it possible to reduce image-size before uploading til eg. imbb?
I got some screenshot from a 2016p resolution movie and the image size is in the area from 28 to 31 MB. I then get a 400 error.

![ee99f7fbcc807f0c0bdb293ff7329e0b](https://user-images.githubusercontent.com/89098574/145084756-6b4e1bd4-2154-474d-a2ec-b2650bc2d47d.png)

**Comment by ryelogheat** [comment]
You can try uploading to [freeimage](https://freeimage.host/l) which has a 64MB file size limit.

 Or if you just want to reduce the filesize of screenshots you might be able to tweak the ffmpeg settings in this line to make that happen https://github.com/ryelogheat/xpbot/blob/de607ea23dd42d8329d1337b79eda052b17b859c/images/upload_screenshots.py#L186 
I'm not super familiar with ffmpeg commands so I don't know how to reduce the file size atm but if you do then feel free to edit that line in your local copy

**Comment by ryelogheat** [comment]
You could probably just create a temporary python file for testing & import this module `from ffmpy import FFmpeg` then you could run the screenshot command (see previous message) all you want with different ffmpeg options without spamming test screenshots to whichever host

**Comment by ghost** [comment]
I'll give Freeimage a try. It's almost impossible to get an account there, så I just have to use their "open" api.

**Reply by ryelogheat** [reply]
What do you mean by "its almost impossible to get an account there"? I signed up a while ago without any issues & they still have a signup form. You should be able to use this API key though 
`6d207e02198a847aa98d0a2a901485a5`
That is the key ^^ they show on their [API details](https://freeimage.host/page/apil) page & interestingly enough its the same one you get if do create an account. So it really doesn't matter if you have a freeimage account or not.

**Reply by ghost** [reply]
The confirmation mail never arrives, hence it's impossible to activate your account.
Yes it seems they have one key to rule'em all. he he
